monai/inferers/utils.py

```diff
@@ -262,12 +262,17 @@ def sliding_window_inference(
 
     # account for any overlapping sections
     for ss in range(len(output_image_list)):
-        output_image_list[ss] = (output_image_list[ss] / count_map_list.pop(0)).to(compute_dtype)
+        output_image_list[ss] = output_image_list[ss].detach()
+        _map = count_map_list.pop(0)
+        for _i in range(output_image_list[ss].shape[1]):
+            output_image_list[ss][:, _i : _i + 1, ...] /= _map
+        output_image_list[ss] = output_image_list[ss].to(compute_dtype)
 
     # remove padding if image_size smaller than roi_size
     for ss, output_i in enumerate(output_image_list):
-        if torch.isnan(output_i).any() or torch.isinf(output_i).any():
-            warnings.warn("Sliding window inference results contain NaN or Inf.")
+        for _i in range(output_i.shape[1]):
+            if torch.isnan(output_i[:, _i, ...]).any() or torch.isinf(output_i[:, _i, ...]).any():
+                warnings.warn(f"{_i}th channel of sliding window inference results contains NaN or Inf.")
```

Comment:
shall we remove this check and warning? this is inefficient and the warning message is not very informative in general

Reply:
I think so too.

Reply:
it is fine to me. but not sure for other applications.

Reply:
@wyli I have removed this section.